src/response.rs

```diff
@@ -6,12 +6,20 @@ use crate::http;
 #[derive(Debug, Clone)]
 pub(crate) struct ResponseUri(pub http::Uri);
 
+#[derive(Debug, Clone)]
+pub(crate) struct RedirectHistory(pub Vec<Uri>);
+
 /// Extension trait for `http::Response<Body>` objects
 ///
 /// Allows the user to access the `Uri` in http::Response
 pub trait ResponseExt {
     /// The Uri we ended up at. This can differ from the request uri when we have followed redirects.
     fn get_uri(&self) -> &Uri;
+
+    /// The full history of uris, including the request and final uri.
+    ///
+    /// Returns None when `redirect_history` is false.
+    fn get_redirect_history(&self) -> Option<&Vec<Uri>>;
```

Comment:
Let's change this to `-> Option<&[Uri]>`. The user can't really do anything with the fact that it is a Vec, so it's better for us to consider that an internal detail. In the future we might consider using SmallVec/TinyVec and store it on the stack instead.

What do you think about always returning `-> &[Uri]` and always return empty if `save_redirect_history` is off? It would be easier for the user to not have to unwrap the optional. Possibly confusing that it's always empty if you failed to read the docs.

Reply:
> What do you think about always returning -> &[Uri] and always return empty if save_redirect_history is off? It would be easier for the user to not have to unwrap the optional. Possibly confusing that it's always empty if you failed to read the docs.

I did the Option since I figured there could be confusion with `get_redirect_history` returning an empty.
 Vec/Slice. Also, since `extensions().get` returns an Option it is simple to just map it to `Some(Slice)` when `save_redirect_history` is true, or return the `None`.

Reply:
Alright. Let's leave it like that.